examples/nucleo_l432kc/spi_dma/main.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2020, Mike Wolfram
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/board.hpp>
+
+using Mosi = modm::platform::GpioOutputB5;
+using Miso = modm::platform::GpioInputB4;
+using Sck = modm::platform::GpioOutputB3;
+using DmaRx = modm::platform::Dma1::Channel2;
+using DmaTx = modm::platform::Dma1::Channel3;
+using Spi = modm::platform::SpiMaster1_Dma<DmaRx, DmaTx>;
+
+int main()
+{
+	Board::initialize();
+
+	// Enable DMA1 controller
+    modm::platform::Dma1::enable();
+	// Enable SPI1
+    Spi::connect<Mosi::Mosi, Miso::Miso, Sck::Sck>();
+    Spi::initialize<Board::SystemClock, 10_MHz>();
+
+	while (true)
+	{
+		uint8_t sendBuffer[13] { "data to send" };
+		uint8_t receiveBuffer[13];
```

Comment:
Trailing 0.

Reply:
You're right. but: Arggh.

```cpp
uint8_t buf[2] {1, 2};  // ok
uint8_t buf[2] {'a','b'};  // ok
uint8_t buf[2] {"a", "b"};  // not ok
uint8_t buf[2] {"b"};  // ok
```

So this would save us one byte of memory but is unreadable: :unamused: 
```cpp
uint8_t sendBuffer[12] { 'd', 'a', 't', 'a', ' ', 't', 'o', ' ', 's', 'e', 'n', 'd' };
```

Reply:
Come on, it's an example, who cares about 1 byte? :-)

And initializing with 12 doesn't work.

Reply:
I completely agree with your version. I was not aware of the trailing 0 during array string initialization. (Therefore I had marked the conversation as "resolved".)